src/components/Button/index.tsx

```diff
@@ -92,8 +91,11 @@ export const Button = forwardRef(
     ref
   ) => {
     const Element = baseElement({ href, onClick: true, as: Link });
+    interface IconProps {
```

Comment:
We can go back and make the decision to make these all types later if we so choose.